### PR TITLE
udevil: refactor and fix #35650

### LINKED ISSUE #35650: devmon as distributed with udevil 0.4.4 is broken
## Issue description

- systemd service isn't referencing the correct binary
- investigate is devmon requires suid

## Technical details

 - system: `"x86_64-linux"`
 - host os: `Linux 4.15.4, NixOS, 18.03.git.66cd46f2d44M (Impala)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 1.11.16`
 - channels(root): `"nixos-18.03pre121670.4cbbecc85db"`
 - channels(peter): `""`
 - nixpkgs: `/home/peter/src/active/nixpkgs_unstable`

### PR DESCRIPTION
**Pull request by Artturin**
###### Motivation for this change
Closes #35650

~~i also updated it to the same commit that arch uses https://github.com/archlinux/svntogit-community/blob/7c87c0d9707926dd679acf30e4624ec181687331/trunk/PKGBUILD~~
###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Relase notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by peterhoeg** [comment]
Did you check out the suid bit?

File: pkgs/applications/misc/udevil/default.nix [hunk 1]
```diff
@@ -1,12 +1,20 @@
...
+{ lib, stdenv, fetchFromGitHub, intltool, glib, pkg-config, udev, util-linux, acl }:
+
+stdenv.mkDerivation rec {
+  pname = "udevil";
+  version = "0.4.4";
+
+  src = fetchFromGitHub {
+    owner = "IgnorantGuru";
+    repo = "udevil";
+    rev = "77a61806fadbf19f02072df1aa67191fc375c35d";
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/udevil/default.nix:10]:
```suggestion
    rev = version;
```

**Comment by Artturin** [comment]
> Did you check out the suid bit?

only udevil is set the suid bit in the Makefile.in `$(INSTALL_DATA) -o root -g root -m 4755 -D $(INSTALL_SRC)/udevil $(INSTALL_DIR)/udevil`
udevmon is run via a systemd system service

**Comment by Artturin** [comment]
i dont use udevil so someone who uses it can test

File: pkgs/applications/misc/udevil/default.nix [hunk 2]
```diff
@@ -19,11 +27,18 @@ stdenv.mkDerivation {
...
   patches = [ ./device-info-sys-stat.patch ];
-  meta = {
+
+  meta = with lib; {
     description = "A command line Linux program which mounts and unmounts removable devices without a password, shows device info, and monitors device changes";
     homepage = "https://ignorantguru.github.io/udevil/";
-    platforms = lib.platforms.linux;
-    license = lib.licenses.gpl3;
+    platforms = platforms.linux;
+    license = licenses.gpl3;
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/udevil/default.nix:42]: Do you know if the license is gpl3Only or gpl3Plus?

Comment (Artturin) [reply pkgs/applications/misc/udevil/default.nix:42]: https://github.com/IgnorantGuru/udevil/blob/ce9f1c5f811e34913b050aed193a39a63ac81e73/COPYING#L640
gpl3Plus

You should make a repo where you list common things to check such as the license

**Comment by peterhoeg** [comment]
@Artturin , why are we adding the devmon service? All this does is to (try to) spawn an invalid instance named `devmon@multi-user`?

**Comment by Artturin** [comment]
> @Artturin , why are we adding the devmon service? All this does is to (try to) spawn an invalid instance named `devmon@multi-user`?

oh there's a devmon module already i didn't check
https://github.com/NixOS/nixpkgs/blob/nixos-unstable/nixos/modules/services/misc/devmon.nix
Does it work?

why is there even a udevil module 
https://github.com/NixOS/nixpkgs/blob/nixos-unstable/nixos/modules/programs/udevil.nix
all it does is give it the SUID bit which according to the arch wiki isn't needed
https://wiki.archlinux.org/title/Udisks#Mount_helpers
```
Note: devmon only uses udisks or udisks2 for mounting (in this order) if udevil or pmount miss the SUID permission. To remove this permission, run chmod -s /usr/bin/udevil as root.
```

**Pull request merged**